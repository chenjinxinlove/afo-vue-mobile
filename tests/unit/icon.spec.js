import Icon from '../../src/icon/index'
import Vue from 'vue'
import sinon from 'sinon'
import Demo from '../../src/icon/demo'
import { shallowMount } from '@vue/test-utils'

describe('Icon.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Icon)
    expect(Vue.component(Icon.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // 测试是否可以正确的显示内容
  it('should render correct contents', () => {
    const wrapper = shallowMount(Icon, {
      propsData: {type: 'select'}
    })
    expect(wrapper.find('i').classes()).toContain('afo-icon-select')
  })
  // 测试props
  it('props', () => {
    const wrapper = shallowMount(Icon, {
      propsData: {
        type: 'select',
        color: 'red',
        size: '24px'
      }
    })
    expect(wrapper.attributes().style).toBe('color: red; font-size: 24px;')
    expect(wrapper.html()).toBe('<i class="afo-icon afo-icon-select" style="color: red; font-size: 24px;"></i>')
  })
  // 测试事件可以被处理
  it('events-called', () => {
    const clickHandler = sinon.stub()
    const wrapper = shallowMount(Icon)
    wrapper.vm.$on('onClick', clickHandler)
    wrapper.trigger('click')
    expect(clickHandler.called).toBe(true)
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(shallowMount(Demo).html()).toMatchSnapshot()
  })
})
