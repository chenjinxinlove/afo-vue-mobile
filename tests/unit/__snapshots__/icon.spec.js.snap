// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.vue demo snapshot 1`] = `
<div>
  <div>
    <i class="afo-icon afo-icon-eye-invisible" style="font-size: 32px;"></i> <span>eye-invisible</span></div>
  <div>
    <i class="afo-icon afo-icon-eye-visible" style="font-size: 32px;"></i> <span>eye-visible</span></div>
  <div>
    <i class="afo-icon afo-icon-square-right" style="font-size: 32px;"></i> <span>square-right</span></div>
  <div>
    <i class="afo-icon afo-icon-select" style="font-size: 32px;"></i> <span>select</span></div>
  <div>
    <i class="afo-icon afo-icon-pulldown" style="font-size: 32px;"></i> <span>pulldown</span></div>
  <div>
    <i class="afo-icon afo-icon-pullup" style="font-size: 32px;"></i> <span>pullup</span></div>
  <div>
    <i class="afo-icon afo-icon-more" style="font-size: 32px;"></i> <span>more</span></div>
  <div>
    <i class="afo-icon afo-icon-back" style="font-size: 32px;"></i> <span>back</span></div>
  <div>
    <i class="afo-icon afo-icon-arrow" style="font-size: 32px;"></i> <span>arrow</span></div>
  <div>
    <i class="afo-icon afo-icon-close" style="font-size: 32px;"></i> <span>close</span></div>
  <div>
    <i class="afo-icon afo-icon-warn" style="font-size: 32px;"></i> <span>warn</span></div>
  <div>
    <i class="afo-icon afo-icon-question" style="font-size: 32px;"></i> <span>question</span></div>
  <div>
    <i class="afo-icon afo-icon-right" style="font-size: 32px;"></i> <span>right</span></div>
  <div>
    <i class="afo-icon afo-icon-wrong" style="font-size: 32px;"></i> <span>wrong</span></div>
  <div>
    <i class="afo-icon afo-icon-info" style="font-size: 32px;"></i> <span>info</span></div>
  <div>
    <i class="afo-icon afo-icon-remove" style="font-size: 32px;"></i> <span>remove</span></div>
  <div>
    <i class="afo-icon afo-icon-add" style="font-size: 32px;"></i> <span>add</span></div>
  <div>
    <i class="afo-icon afo-icon-share" style="font-size: 32px;"></i> <span>share</span></div>
  <div>
    <i class="afo-icon afo-icon-no-wifi" style="font-size: 32px;"></i> <span>no-wifi</span></div>
  <div>
    <i class="afo-icon afo-icon-smile" style="font-size: 32px;"></i> <span>smile</span></div>
  <div>
    <i class="afo-icon afo-icon-sad" style="font-size: 32px;"></i> <span>sad</span></div>
  <div>
    <i class="afo-icon afo-icon-email" style="font-size: 32px;"></i> <span>email</span></div>
  <div>
    <i class="afo-icon afo-icon-game" style="font-size: 32px;"></i> <span>game</span></div>
  <div>
    <i class="afo-icon afo-icon-wifi" style="font-size: 32px;"></i> <span>wifi</span></div>
  <div>
    <i class="afo-icon afo-icon-hot" style="font-size: 32px;"></i> <span>hot</span></div>
  <div>
    <i class="afo-icon afo-icon-notification" style="font-size: 32px;"></i> <span>notification</span></div>
  <div>
    <i class="afo-icon afo-icon-delete" style="font-size: 32px;"></i> <span>delete</span></div>
  <div>
    <i class="afo-icon afo-icon-vip" style="font-size: 32px;"></i> <span>vip</span></div>
  <div>
    <i class="afo-icon afo-icon-mute" style="font-size: 32px;"></i> <span>mute</span></div>
  <div>
    <i class="afo-icon afo-icon-danger" style="font-size: 32px;"></i> <span>danger</span></div>
  <div>
    <i class="afo-icon afo-icon-volume" style="font-size: 32px;"></i> <span>volume</span></div>
  <div>
    <i class="afo-icon afo-icon-bad" style="font-size: 32px;"></i> <span>bad</span></div>
  <div>
    <i class="afo-icon afo-icon-mobile-phone" style="font-size: 32px;"></i> <span>mobile-phone</span></div>
  <div>
    <i class="afo-icon afo-icon-aim" style="font-size: 32px;"></i> <span>aim</span></div>
  <div>
    <i class="afo-icon afo-icon-navigation" style="font-size: 32px;"></i> <span>navigation</span></div>
  <div>
    <i class="afo-icon afo-icon-safe-pay" style="font-size: 32px;"></i> <span>safe-pay</span></div>
  <div>
    <i class="afo-icon afo-icon-tag" style="font-size: 32px;"></i> <span>tag</span></div>
  <div>
    <i class="afo-icon afo-icon-lock" style="font-size: 32px;"></i> <span>lock</span></div>
  <div>
    <i class="afo-icon afo-icon-unlock" style="font-size: 32px;"></i> <span>unlock</span></div>
  <div>
    <i class="afo-icon afo-icon-edit" style="font-size: 32px;"></i> <span>edit</span></div>
  <div>
    <i class="afo-icon afo-icon-scan" style="font-size: 32px;"></i> <span>scan</span></div>
  <div>
    <i class="afo-icon afo-icon-qr-code" style="font-size: 32px;"></i> <span>qr-code</span></div>
  <div>
    <i class="afo-icon afo-icon-calendar" style="font-size: 32px;"></i> <span>calendar</span></div>
  <div>
    <i class="afo-icon afo-icon-time" style="font-size: 32px;"></i> <span>time</span></div>
  <div>
    <i class="afo-icon afo-icon-red-packet" style="font-size: 32px;"></i> <span>red-packet</span></div>
  <div>
    <i class="afo-icon afo-icon-star" style="font-size: 32px;"></i> <span>star</span></div>
  <div>
    <i class="afo-icon afo-icon-setting" style="font-size: 32px;"></i> <span>setting</span></div>
  <div>
    <i class="afo-icon afo-icon-home" style="font-size: 32px;"></i> <span>home</span></div>
  <div>
    <i class="afo-icon afo-icon-credit-card" style="font-size: 32px;"></i> <span>credit-card</span></div>
  <div>
    <i class="afo-icon afo-icon-mall" style="font-size: 32px;"></i> <span>mall</span></div>
  <div>
    <i class="afo-icon afo-icon-microphone" style="font-size: 32px;"></i> <span>microphone</span></div>
  <div>
    <i class="afo-icon afo-icon-search" style="font-size: 32px;"></i> <span>search</span></div>
  <div>
    <i class="afo-icon afo-icon-good" style="font-size: 32px;"></i> <span>good</span></div>
  <div>
    <i class="afo-icon afo-icon-alert" style="font-size: 32px;"></i> <span>alert</span></div>
  <div>
    <i class="afo-icon afo-icon-picture" style="font-size: 32px;"></i> <span>picture</span></div>
  <div>
    <i class="afo-icon afo-icon-message" style="font-size: 32px;"></i> <span>message</span></div>
  <div>
    <i class="afo-icon afo-icon-phone" style="font-size: 32px;"></i> <span>phone</span></div>
  <div>
    <i class="afo-icon afo-icon-location" style="font-size: 32px;"></i> <span>location</span></div>
  <div>
    <i class="afo-icon afo-icon-like" style="font-size: 32px;"></i> <span>like</span></div>
  <div>
    <i class="afo-icon afo-icon-camera" style="font-size: 32px;"></i> <span>camera</span></div>
  <div>
    <i class="afo-icon afo-icon-person" style="font-size: 32px;"></i> <span>person</span></div>
  <div>
    <i class="afo-icon afo-icon-round-border" style="font-size: 32px;"></i> <span>round-border</span></div>
  <div>
    <i class="afo-icon afo-icon-important" style="font-size: 32px;"></i> <span>important</span></div>
  <div>
    <i class="afo-icon afo-icon-ok" style="font-size: 32px;"></i> <span>ok</span></div>
  <div>
    <i class="afo-icon afo-icon-square-border" style="font-size: 32px;"></i> <span>square-border</span></div>
</div>
`;
